app/code/Magento/Catalog/Ui/Component/Listing/Columns/StockStatus.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Catalog\Ui\Component\Listing\Columns;
+
+use Magento\Framework\View\Element\UiComponentFactory;
+use Magento\Framework\View\Element\UiComponent\ContextInterface;
+
+/**
+ * Class StockStatus
+ */
+class StockStatus extends \Magento\Ui\Component\Listing\Columns\Column
+{
+    /**
+     * Product stock status attribute code
+     */
+    const NAME = 'quantity_and_stock_status';
+
+    /**
+     * @var \Magento\CatalogInventory\Api\StockRegistryInterface
+     */
+    protected $stockRegistry;
```

Comment:
Please change visibility to private for all properties that are used only inside the same class

Reply:
changes was provided